### PR TITLE
Add --disable-heartbeat flag for linkerd install|upgrade

### LINKED ISSUE #278: test failure due to data race
```
==================
WARNING: DATA RACE
Read at 0x00c4201f8218 by goroutine 36:
  github.com/runconduit/conduit/controller/api/public.(*mockGrpcServer).Tap()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server_test.go:53 +0x115
  github.com/runconduit/conduit/controller/api/public.(*handler).handleTap()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server.go:158 +0x18b
  github.com/runconduit/conduit/controller/api/public.(*handler).ServeHTTP()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server.go:50 +0x4db
  net/http.serverHandler.ServeHTTP()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:2619 +0xbc
  net/http.(*conn).serve()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:1801 +0x83b
Previous write at 0x00c4201f8218 by goroutine 48:
  github.com/runconduit/conduit/controller/api/public.TestServer.func4()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server_test.go:179 +0xcd
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:746 +0x16c
Goroutine 36 (running) created at:
  net/http.(*Server).Serve()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:2720 +0x37c
  net/http.Serve()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:2323 +0xe2
  github.com/runconduit/conduit/controller/api/public.TestServer.func1()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server_test.go:75 +0x5c
Goroutine 48 (finished) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:789 +0x568
  github.com/runconduit/conduit/controller/api/public.TestServer()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server_test.go:178 +0x58a
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:746 +0x16c
==================
==================
WARNING: DATA RACE
Read at 0x00c420466de0 by goroutine 36:
  errors.(*errorString).Error()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/errors/errors.go:19 +0x3b
  github.com/runconduit/conduit/controller/api/public.writeErrorToHttpResponse()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/proto_over_http.go:71 +0x163
  github.com/runconduit/conduit/controller/api/public.(*handler).handleTap()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server.go:160 +0x1c2
  github.com/runconduit/conduit/controller/api/public.(*handler).ServeHTTP()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server.go:50 +0x4db
  net/http.serverHandler.ServeHTTP()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:2619 +0xbc
  net/http.(*conn).serve()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:1801 +0x83b
Previous write at 0x00c420466de0 by goroutine 48:
  github.com/runconduit/conduit/controller/api/public.TestServer.func4()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/client.go:5 +0x93
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:746 +0x16c
Goroutine 36 (running) created at:
  net/http.(*Server).Serve()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:2720 +0x37c
  net/http.Serve()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:2323 +0xe2
  github.com/runconduit/conduit/controller/api/public.TestServer.func1()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server_test.go:75 +0x5c
Goroutine 48 (finished) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:789 +0x568
  github.com/runconduit/conduit/controller/api/public.TestServer()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server_test.go:178 +0x58a
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:746 +0x16c
==================
=== RUN   TestDeserializePayloadFromReader/Returns_error_when_message_has_fewer_bytes_than_declared_message_size
--- FAIL: TestDeserializePayloadFromReader (0.05s)
    --- PASS: TestDeserializePayloadFromReader/Can_read_message_correctly_based_on_payload_size_correct_payload_size_to_message (0.00s)
    --- PASS: TestDeserializePayloadFromReader/Can_multiple_messages_in_the_same_stream (0.03s)
    --- PASS: TestDeserializePayloadFromReader/Can_write_and_read__marshalled_protobuf_messages (0.01s)
    --- PASS: TestDeserializePayloadFromReader/Can_read_byte_streams_larger_than_Go's_default_buffer_chunk_size (0.02s)
    --- PASS: TestDeserializePayloadFromReader/Returns_error_when_message_has_fewer_bytes_than_declared_message_size (0.00s)
	testing.go:699: race detected during execution of test
```

### LINKED ISSUE #3386: Introduce CLI install option to disable heartbeat
The `DisableHeartBeat` variable introduced by https://github.com/linkerd/linkerd2/pull/3366 allows users to disable the `heartbeat` service, during installation by the Linkerd and Helm CLI. But this config isn't persisted. In particular, when the user runs the `linkerd check` command, the `check` will fail due to the missing `heartbeat` service account:

```
× control plane ServiceAccounts exist
    missing ServiceAccounts: linkerd-heartbeat
    see https://linkerd.io/checks/#l5d-existence-sa for hints
```

The scope of this PR is to introduce a `--disable-heartbeat` option to the `linkerd install` command. This option, defaults to `false`, will be a "persisted flag", which means its value will be persisted in the `linkerd-config` config map, as part of the `install` data. Moreover, the [`healthcheck.expectedServiceAccount()`](https://github.com/linkerd/linkerd2/blob/d3690299090ae87baaafae307081c256a055d4d9/pkg/healthcheck/healthcheck.go#L1036-L1048) function will be updated to include the `linkerd-heartbeat` service account, only if `heartbeat` is enabled.

### PR DESCRIPTION
**Pull request by alpeb**
Fixes #3386

Add `linkerd install|upgrade --disable-heartbeat` flag, and have
`linkerd check` check for the heartbeat's SA only if it's enabled.

Also added those flags into the `linkerd upgrade -h` examples.

### CODE REVIEW

File: test/testdata/check.proxy.golden [hunk 1]
```diff
@@ -22,6 +22,7 @@ linkerd-config
 linkerd-existence
 -----------------
 √ 'linkerd-config' config map exists
+√ control plane ServiceAccounts exist
```
Comment (adleong) [inline test/testdata/check.proxy.golden]: feels weird that this line appears twice but each line has a different meaning.

File: pkg/healthcheck/healthcheck.go [hunk 2]
```diff
@@ -588,6 +588,14 @@ func (hc *HealthChecker) allCategories() []category {
 						return
 					},
 				},
+				{
+					description: "control plane ServiceAccounts exist",
```
[the remark this answers is not in the document]
Comment (alpeb) [reply pkg/healthcheck/healthcheck.go]: Before checking if the heartbeat SA exists we need to know if heartbeat is enabled by looking into linkerd-config which only exists after the control plane has been set up (at the `linkerd install control-plane` stage) and thus not necessarily available when the `LinkerdConfigChecks` checks are called.

I didn't want to expose more details than needed for this specific check, so I kept on using the generic "control plane ServiceAccounts" label. I can be convinced to use a separate label, but I think it's not worth a separate hintAnchor, as the existing one carries all the necessary info about the SAs the system needs (the docs need to be updated though, to include the heartbeat SA in the list).

File: pkg/healthcheck/healthcheck.go [hunk 3]
```diff
@@ -1093,7 +1103,7 @@ func (hc *HealthChecker) checkClusterRoleBindings(shouldExist bool) error {
 	return checkResources("ClusterRoleBindings", objects, hc.expectedRBACNames(), shouldExist)
 }
 
-func (hc *HealthChecker) checkServiceAccounts(shouldExist bool) error {
+func (hc *HealthChecker) checkServiceAccounts(onlyHeartbeat bool) error {
```
[the remark this answers is not in the document]
Comment (alpeb) [reply pkg/healthcheck/healthcheck.go]: Yeah that should be cleaner :+1:

File: pkg/healthcheck/healthcheck.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (adleong) [reply pkg/healthcheck/healthcheck.go]: That all makes sense to me.  A different label with the same hintAnchor seems totally reasonable.

**Comment by alpeb** [comment]
@adleong I've just pushed a commit with your suggestions, for your perusal :mag:

**Pull request merged**